sha3_api.h

```diff
@@ -31,7 +32,18 @@ extern "C"
 
   /* incremental interfece */
   void sha3_init(sha3_var_t *c,
-                 size_t md_sz); /* md_sz = hash output in bytes */
+                 size_t md_sz) /* md_sz = hash output in bytes */
+  __contract__(
+    requires(memory_no_alias(c, sizeof(sha3_var_t)))
+    requires(md_sz <= 49)
+    assigns(object_whole(c))
+    ensures(forall(i, 0, 25, c->st.d[i] == 0))
+    ensures(c->md_sz <= 49)
+    ensures(c->r_sz <= 200)
+    ensures(c->r_sz >= 0)
+    ensures(c->pt == 0)
```

Comment:
The question here becomes if these post-conditions are actually needed for any of the functions calling this function. It is hard to tell at this moment what is needed and what is not needed. 
I think it is easier to leave out the post-conditions for now.
We can always revisit it later and strengthen it.